presto-raptor/src/test/java/com/facebook/presto/raptor/metadata/TestShardPredicate.java

```diff
@@ -98,14 +98,42 @@ public void testRangeShardUuidPredicate()
     }
 
     @Test
-    public void testBucketNumber()
+    public void testBucketNumberSingleRange()
     {
         TupleDomain<RaptorColumnHandle> tupleDomain = withColumnDomains(ImmutableMap.of(
                 bucketNumberColumnHandle("test"),
                 create(SortedRangeSet.copyOf(INTEGER, ImmutableList.of(equal(INTEGER, 1L))), false)));
 
         ShardPredicate shardPredicate = ShardPredicate.create(tupleDomain, true);
-        assertEquals(shardPredicate.getPredicate(), "(bucket_number >= ? OR bucket_number IS NULL) AND (bucket_number <= ? OR bucket_number IS NULL)");
+        assertEquals(shardPredicate.getPredicate(), "((bucket_number >= ? OR bucket_number IS NULL) AND (bucket_number <= ? OR bucket_number IS NULL))");
+    }
+
+    @Test
+    public void testBucketNumberMultipleRanges()
+    {
+        TupleDomain<RaptorColumnHandle> tupleDomain = withColumnDomains(ImmutableMap.of(
+                bucketNumberColumnHandle("test"),
+                create(SortedRangeSet.copyOf(INTEGER, ImmutableList.of(equal(INTEGER, 1L), equal(INTEGER, 3L))), false)));
+
+        ShardPredicate shardPredicate = ShardPredicate.create(tupleDomain, true);
+        assertEquals(shardPredicate.getPredicate(),
+                "((bucket_number >= ? OR bucket_number IS NULL) AND (bucket_number <= ? OR bucket_number IS NULL)" +
```

Comment:
Per the comment above about parenthesis, this is
```
(x1) AND (x2) OR (y1) AND (y2)
```
We want the precedence to be explicit:
```
((x1) AND (x2)) OR ((y1) AND (y2))
```